SlicerDWD/SlicerDWD.py

```diff
@@ -20,12 +20,30 @@
 
 import sklearn.metrics
 
+import importlib
+
+DWD_VERSION = '1.0.2'
+
 try:
     import dwd
 except ModuleNotFoundError:
-    slicer.util.pip_install("dwd")
+    slicer.util.pip_install('dwd=={}'.format(DWD_VERSION))
     import dwd
 
+if dwd.__version__ != DWD_VERSION:
+    slicer.util.pip_install('dwd=={}'.format(DWD_VERSION))
+    importlib.reload(dwd)
+
+    def request_restart():
+        if slicer.util.confirmYesNoDisplay(
+            'SlicerDWD has updated Python dependencies. '
+            'It is recommended to restart Slicer. Restart now?'
+        ):
+            slicer.util.restart()
+            slicer.util.exit()
+
+    qt.QTimer.singleShot(0, request_restart)
```

Comment:
@jcfr I looked in `slicer.util` and `QApplication` (`slicer.app`) and I don't see a clean way to restart the app here because this code happens while modules are being initialized.

At this point, the Slicer widget is not yet instantiated so `slicer.util.confirmYesNoDisplay` does not work directly. Using a single shot timer resolves this and allows the widget to appear.

However since Slicer is not done initializing, `slicer.util.exit()` fails; the original instance of Slicer continues its initialization and launches as normal. The preceding `slicer.util.restart()` simply launches a second instance of Slicer.

I experimented a bit with connecting `slicer.app.onApplicationStateChanged` to `request_restart`; so the prompt would appear only after initialization completes (the state becomes `ApplicationActive` or `ApplicationHidden`). While this works, it doesn't feel quite right... Wondering if you have any insight @jcfr.

```python
def request_restart():
    if slicer.util.confirmYesNoDisplay(
        'SlicerDWD has updated Python dependencies. '
        'It is recommended to restart Slicer. Restart now?'
    ):
        slicer.util.restart()
        slicer.util.exit()
    else:
        slicer.app.applicationStateChanged.disconnect(request_restart)
        
slicer.app.applicationStateChanged.connect(request_restart)
```